internal/encoded.go

```diff
@@ -25,7 +25,20 @@
 package internal
 
 import (
+	"encoding/json"
+	"errors"
+	"fmt"
 	"reflect"
+
+	commonpb "go.temporal.io/temporal-proto/common"
+)
+
+const (
+	metadataEncoding     = "encoding"
+	metadataEncodingRaw  = "raw"
+	metadataEncodingJSON = "json"
```

Comment:
What about adding primitive types? At least numbers and strings? Now we are wrapping them as Json.